pkg/util/workqueue/metrics.go

```diff
@@ -0,0 +1,137 @@
+package workqueue
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type queueMetrics interface {
+	add(item t)
+	get(item t)
+	done(item t)
+}
+
+type defaultQueueMetrics struct {
+	depth                prometheus.Gauge
+	adds                 prometheus.Counter
+	latency              prometheus.Summary
+	workDuration         prometheus.Summary
+	addTimes             map[t]time.Time
+	processingStartTimes map[t]time.Time
+}
+
+func newQueueMetrics(name string) queueMetrics {
+	var ret *defaultQueueMetrics
```

Comment:
> nit: preference for:

I think that's actually subtly different.  This implementation returns a `nil` `defaultQueueMetrics`, which then handles the case where its method target (the `*defaultQueueMetrics` is `nil`.  If you return `nil` directly, you'll end up having a nil value for interface and panic in the method call, right?